### PR TITLE
SONAR-6740 Too many error logs when server fails to start

### PR DESCRIPTION
**Pull request by simonbrandhof**
Ruby on Rails must not be initialized when Java components fail to start. 
It allows to not display all the additional stacktraces generated by the
subsequent Rails errors.

### CODE REVIEW

**Comment by julienlancelot** [comment]
Rails logs are not more generated when the server fails to start, but there are still some logs on elasticsearch threads that still alive.

**Comment by julienlancelot** [comment]
Great job ! ES logs has disappeared

File: server/sonar-server/src/main/java/org/sonar/server/app/TomcatContexts.java [hunk 1]
```diff
@@ -19,34 +19,73 @@
...
+      fs.createOrCleanupDir(dir);
+    } catch (IOException e) {
+      throw new IllegalStateException(format("Fail to create or clean-up directory %s", dir.getAbsolutePath()), e);
+    }
 
-      StandardContext context = (StandardContext) tomcat.addWebapp(ROOT_CONTEXT_PATH, webappPath(props));
+    return addContext(tomcat, contextPath, dir);
+  }
+
+  private StandardContext addContext(Tomcat tomcat, String contextPath, File dir) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/app/TomcatContexts.java]: MAJOR Make "addContext" a "static" method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Pull request closed without merging**